Module naming and versioning blog post (#10)
cekit/cekit.io: content/blog/2019/07/module-naming-and-versioning-guidelines.adoc

---
title: "Module naming and versioning guidelines"
author: "Marek Goldmann"
date: 2019-07-11
tags: ["guideline"]
draft: false
---

As part of the upcoming 3.3.0 release preparation -- module documentation was updated
in several places. Among other things module naming and versioning pages
were added. Below you can find a summary of our suggestions.

> You can already browse link:https://docs.cekit.io/en/develop/[development version of documentation].

Reviewer: ```suggestion
You can already browse the link:https://docs.cekit.io/en/develop/[development version of the documentation].
```

> It will became main documentation after 3.3.0 release.

Reviewer: ```suggestion
It will become the main documentation after the 3.3.0 release.
```

NOTE: You are free to use your own module name and versioning scheme, but standardizing on the same
format will make it easier to reuse modules prepared by someone else.

== Module naming

> See link:https://docs.cekit.io/en/develop/guidelines/modules/naming.html[development version of documentation]
related to module naming.

> We suggest to use link:https://en.wikipedia.org/wiki/Reverse_domain_name_notation[reverse domain convention].

Reviewer: ```suggestion
We suggest using link:https://en.wikipedia.org/wiki/Reverse_domain_name_notation[reverse domain convention].
```

You can see an example below:

{{< highlight yaml >}}
name: "org.jboss.container.openjdk"
version: "11"
description: "OpenJDK 11 module"
{{< /highlight >}}

In the image descriptor's module's installation section we could see something similar to this:

{{< highlight yaml >}}
modules:
    install:
        - name: "org.jboss.container.openjdk"
          version: "11"
        - name: "org.projecta.feature"
          version: "1.2"
        - name: "org.projectb.feature"
          version: "1.12.1"
{{< /highlight >}}

> Here are the main benefits of using such approach:

Reviewer: ```suggestion
Here are the main benefits of using such an approach:
```
